diff -ruN libzypp-9.10.2.orig/zypp/media/MediaCurl.cc libzypp-9.10.2/zypp/media/MediaCurl.cc
--- libzypp-9.10.2.orig/zypp/media/MediaCurl.cc	2011-09-06 14:58:01.669546652 +0300
+++ libzypp-9.10.2/zypp/media/MediaCurl.cc	2011-09-06 14:59:11.788051355 +0300
@@ -554,6 +554,9 @@
       ZYPP_RETHROW(e);
   }
 
+  if ( _url.getQueryParam( "proxy" ) == "_none_" )
+      SET_OPTION(CURLOPT_NOPROXY, "*");
+
   // if the proxy was not set by url, then look
   if ( _settings.proxy().empty() )
   {
@@ -647,12 +650,6 @@
         SET_OPTION(CURLOPT_PROXYUSERPWD, proxyuserpwd.c_str());
     }
   }
-  else
-  {
-#if CURLVERSION_AT_LEAST(7,19,4)
-      SET_OPTION(CURLOPT_NOPROXY, "*");
-#endif
-  }
 
   /** Speed limits */
   if ( _settings.minDownloadSpeed() != 0 )
diff -ruN libzypp-9.10.2.orig/zypp/parser/RepoFileReader.cc libzypp-9.10.2/zypp/parser/RepoFileReader.cc
--- libzypp-9.10.2.orig/zypp/parser/RepoFileReader.cc	2011-09-06 14:58:01.652546286 +0300
+++ libzypp-9.10.2/zypp/parser/RepoFileReader.cc	2011-09-06 14:58:11.196751101 +0300
@@ -90,7 +90,8 @@
                url.setQueryParam("proxy", what[1]);
                url.setQueryParam("proxyport", what[2]);
               }
-            }
+            } else
+              url.setQueryParam("proxy", "_none_");
           } else
             ERR << "Unknown attribute in [" << *its << "]: " << it->second << " ignored" << endl;
         }
