From ec0df2f6360e498e0aa3251a5a926d67d74c5434 Mon, 28 May 2018 15:34:57 +0200
From: Michael Andres <ma@suse.de>
Date: Sat, 25 Jan 2014 17:05:48 +0100
Subject: [PATCH] Get pre/posttrans scripts from RpmHeader


diff --git a/zypp/target/rpm/RpmHeader.cc b/zypp/target/rpm/RpmHeader.cc
index 45e4b90..c638dd7 100644
--- a/zypp/target/rpm/RpmHeader.cc
+++ b/zypp/target/rpm/RpmHeader.cc
@@ -751,59 +751,44 @@
 std::string RpmHeader::tag_os() const
 {
   return string_val( RPMTAG_OS );
+
 }
 
-///////////////////////////////////////////////////////////////////
-//
-//
-//        METHOD NAME : RpmHeader::tag_prein
-//        METHOD TYPE : std::string
-//
-//        DESCRIPTION :
-//
 std::string RpmHeader::tag_prein() const
-{
-  return string_val( RPMTAG_PREIN );
-}
+{ return string_val( RPMTAG_PREIN ); }
 
-///////////////////////////////////////////////////////////////////
-//
-//
-//        METHOD NAME : RpmHeader::tag_postin
-//        METHOD TYPE : std::string
-//
-//        DESCRIPTION :
-//
+std::string RpmHeader::tag_preinprog() const
+{ return string_val( RPMTAG_PREINPROG ); }
+
 std::string RpmHeader::tag_postin() const
-{
-  return string_val( RPMTAG_POSTIN );
-}
+{ return string_val( RPMTAG_POSTIN ); }
 
-///////////////////////////////////////////////////////////////////
-//
-//
-//        METHOD NAME : RpmHeader::tag_preun
-//        METHOD TYPE : std::string
-//
-//        DESCRIPTION :
-//
+std::string RpmHeader::tag_postinprog() const
+{ return string_val( RPMTAG_POSTINPROG ); }
+
 std::string RpmHeader::tag_preun() const
-{
-  return string_val( RPMTAG_PREUN );
-}
+{ return string_val( RPMTAG_PREUN ); }
 
-///////////////////////////////////////////////////////////////////
-//
-//
-//        METHOD NAME : RpmHeader::tag_postun
-//        METHOD TYPE : std::string
-//
-//        DESCRIPTION :
-//
+std::string RpmHeader::tag_preunprog() const
+{ return string_val( RPMTAG_PREUNPROG ); }
+
 std::string RpmHeader::tag_postun() const
-{
-  return string_val( RPMTAG_POSTUN );
-}
+{ return string_val( RPMTAG_POSTUN ); }
+
+std::string RpmHeader::tag_postunprog() const
+{ return string_val( RPMTAG_POSTUNPROG ); }
+
+std::string RpmHeader::tag_pretrans() const
+{ return string_val( RPMTAG_PRETRANS ); }
+
+std::string RpmHeader::tag_pretransprog() const
+{ return string_val( RPMTAG_PRETRANSPROG ); }
+
+std::string RpmHeader::tag_posttrans() const
+{ return string_val( RPMTAG_POSTTRANS ); }
+
+std::string RpmHeader::tag_posttransprog() const
+{ return string_val( RPMTAG_POSTTRANSPROG ); }
 
 ///////////////////////////////////////////////////////////////////
 //
diff --git a/zypp/target/rpm/RpmHeader.h b/zypp/target/rpm/RpmHeader.h
index 02dd8d1..eae75ca 100644
--- a/zypp/target/rpm/RpmHeader.h
+++ b/zypp/target/rpm/RpmHeader.h
@@ -145,9 +145,17 @@
   std::string tag_url()          const;
   std::string tag_os()           const;
   std::string tag_prein()        const;
+  std::string tag_preinprog()    const;
   std::string tag_postin()       const;
+  std::string tag_postinprog()   const;
   std::string tag_preun()        const;
+  std::string tag_preunprog()    const;
   std::string tag_postun()       const;
+  std::string tag_postunprog()   const;
+  std::string tag_pretrans()     const;
+  std::string tag_pretransprog() const;
+  std::string tag_posttrans()    const;
+  std::string tag_posttransprog()const;
   std::string tag_sourcerpm()    const;
 
   /** just the list of names  */
