From 4f7310065d16cf6e777b8e25c5fba4673d00169b Wed, 27 Jun 2018 18:17:24 +0200
From: Niels Breet <niels.breet@jolla.com>
Date: Wed, 2 Sep 2015 18:51:28 +0300
Subject: [PATCH] [libzypp] Set download.use_deltarpm.always=true . Fixes JB#31671


diff --git a/zypp.conf b/zypp.conf
index 4abb5f2..93cfa03 100644
--- a/zypp.conf
+++ b/zypp.conf
@@ -215,7 +215,7 @@
 ##
 ## This option has no effect unless download.use_deltarpm is set true.
 ##
-#  download.use_deltarpm.always = false
+download.use_deltarpm.always = true
 
 ##
 ## Hint which media to prefer when installing packages (download vs. CD).
