diff --git a/zypp.conf b/zypp.conf
index dfe7f64..5292843 100644
--- a/zypp.conf
+++ b/zypp.conf
@@ -199,7 +199,7 @@
 ##
 ## This option has no effect unless download.use_deltarpm is set true.
 ##
-#  download.use_deltarpm.always = false
+download.use_deltarpm.always = true
 
 ##
 ## Hint which media to prefer when installing packages (download vs. CD).
@@ -427,6 +427,16 @@
 # solver.checkSystemFile = /etc/zypp/systemCheck
 
 ##
+## This directory can contain files that contain requirements/conflicts
+## which fulfill the needs of a running system (see checkSystemFile).
+##
+## Files are read in alphabetical order.
+##
+## Default value: {configdir}/systemCheck.d
+##
+# solver.checkSystemFileDir = /etc/zypp/systemCheck.d
+
+##
 ## When committing a dist upgrade (e.g. 'zypper dup') a solver testcase
 ## is written to /var/log/updateTestcase-<date>. It is needed in bugreports.
 ## This option returns the number of testcases to keep on the system. Old
@@ -595,7 +505,7 @@ multiversion.kernels = latest,latest-1,running
 ## Valid values:  boolean
 ## Default value: no
 ##
-# rpm.install.excludedocs = no
+rpm.install.excludedocs = yes
 
 ##
 ## Location of history log file.
 
