From 069567431fc23802f07b5167aa4bd169210bd521 Wed, 27 Jun 2018 18:17:27 +0200
From: Niels Breet <niels.breet@jolla.com>
Date: Wed, 2 Sep 2015 18:51:53 +0300
Subject: [PATCH] [libzypp] Set rpm.install.excludedocs = yes. Save space on mobile devices. Fixes JB#31669


diff --git a/zypp.conf b/zypp.conf
index 93cfa03..1cd12a6 100644
--- a/zypp.conf
+++ b/zypp.conf
@@ -640,7 +640,7 @@
 ## Valid values:  boolean
 ## Default value: no
 ##
-# rpm.install.excludedocs = no
+rpm.install.excludedocs = yes
 
 ##
 ## Location of history log file.
