From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Niels Breet <niels.breet@jolla.com>
Date: Tue, 27 Oct 2020 17:01:42 +0200
Subject: [PATCH] Disable docs building with force

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b0d19a2eef60cb3f15fa30265433dd247bbb2e7f..c143d06a2d8d4f9dd6c784413f518f305ba30c13 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -393,7 +393,7 @@ ADD_SUBDIRECTORY( zypp-curl )
 ADD_SUBDIRECTORY( zypp-tui )
 ADD_SUBDIRECTORY( zypp )
 ADD_SUBDIRECTORY( tools )
-ADD_SUBDIRECTORY( doc )
+# ADD_SUBDIRECTORY( doc )
 
 IF ( ENABLE_BUILD_TRANS )
   ADD_SUBDIRECTORY( po )
